### DISCUSSION TITLE
How to select a geo-server for building iOS applications on Codemagic?

### DISCUSSION THREAD

**Discussion by lukyanenko01** [codemagic-ci-cd/codemagic-docs#2740] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

iOS (Swift, Objective-C)

### Steps to reproduce

	1.	Initiate a new build for an iOS application.
	2.	There doesn’t seem to be an option to select a specific geo-location server for the build process.
	3.	I’m looking for a way to select a server based on geographic location (e.g., Europe, US) to optimize build times and comply with data storage regulations.

### Expected results

I expect to have an option or feature in Codemagic to select a geo-server for the iOS build process to ensure faster build times and data localization compliance.

### Actual results

Currently, I don’t see any option to choose the geographic location of the server where the iOS build takes place. This may lead to slower build times or regulatory concerns regarding data storage.

### Build id (optional)

N/A

**Comment by himesh-cm** [comment]
Hey @lukyanenko01, At the moment, it is not possible to trigger a build from a specific location. If you have custom requirements, I encourage you to read more about custom and dedicated hosts [here]( https://docs.codemagic.io/billing/pricing/#dedicated-hosts).
Feel free to contact us with your specific requirements [here](https://codemagic.io/contact/) and someone from the team will connect with you to take it forward.

**Comment by himesh-cm** [comment]
While I’m closing our discussion for now, if you have any other questions or suggestions, feel free to reply back or start a new discussion anytime. Have a nice day!
